### PR TITLE
Overhaul of list / detail / edit /create view

### PR DESCRIPTION
**Pull request by chraker**
- The new variant lies in customer, check it out

### CODE REVIEW

File: frontend/web/src/app/components/customer-list/customer-list.component.ts [hunk 1]
```diff
@@ -100,10 +101,10 @@ export class CustomerListComponent implements OnInit, AfterViewInit {
   }
 
   private prepareTriggers() {
-    /*let _self = this;
+    let _self = this;
     (<any>this.table).on('click-row.bs.table', function (row, $element) {
```
Comment (chraker) [inline frontend/web/src/app/components/customer-list/customer-list.component.ts]: MINOR Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead. [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS4137)

File: frontend/web/src/app/components/customer-list/customer-list.component.ts [hunk 2]
```diff
@@ -100,10 +101,10 @@ export class CustomerListComponent implements OnInit, AfterViewInit {
...
-      _self.router.navigate([_self.router.url, $element.id]);
-    })*/
+      _self.selectedCustomer =  _self.customers.filter(customer => {if(customer.id ==  $element.id)return customer})[0];
```
Comment (chraker) [inline frontend/web/src/app/components/customer-list/customer-list.component.ts]: MAJOR == should be === [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS1440)

File: frontend/web/src/app/model/order.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+export class Customer {
+
+  public id: number;
+  public customer: number;
+
+  constructor(values: Object = {}) {
+    if (!values)
```
Comment (chraker) [inline frontend/web/src/app/model/order.ts]: CRITICAL if statements must be braced [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS121)

File: frontend/web/src/app/model/order.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+      return null;
+    Object.assign(this, values);
+  }
+
+  static create(json: any) {
+    if (json)
```
Comment (chraker) [inline frontend/web/src/app/model/order.ts]: CRITICAL if statements must be braced [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS121)

File: frontend/web/src/app/model/order.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+      return new Customer(json);
+    else
```
Comment (chraker) [inline frontend/web/src/app/model/order.ts]: CRITICAL else statements must be braced [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS121)

File: frontend/web/src/app/services/physical-part/physical-part-mock.service.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  }
+
+  getAll(): Observable<PhysicalPart[]> {
+    return Observable.of(this.data)
+  }
+  get(id: number): Observable<PhysicalPart> {
+    return Observable.of( this.data.filter(e => {if(e.id == id)return e})[0]);
+  }
+  delete(id: number) {
+    this.data = this.data.filter(e => {if(e.id != id)return e});
```
Comment (chraker) [inline frontend/web/src/app/services/physical-part/physical-part-mock.service.ts]: MAJOR != should be !== [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS1440)

File: frontend/web/src/app/services/physical-part/physical-part-mock.service.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  ];
+
+  constructor() {
+  }
+
+  getAll(): Observable<PhysicalPart[]> {
+    return Observable.of(this.data)
+  }
+  get(id: number): Observable<PhysicalPart> {
+    return Observable.of( this.data.filter(e => {if(e.id == id)return e})[0]);
```
Comment (chraker) [inline frontend/web/src/app/services/physical-part/physical-part-mock.service.ts]: MAJOR == should be === [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS1440)

File: frontend/web/src/app/services/physical-print/physical-print-mock.service.ts [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+  ];
+
+  constructor() {
+  }
+
+  getAll(): Observable<PhysicalPrint[]> {
+    return Observable.of(this.data)
+  }
+  get(id: number): Observable<PhysicalPrint> {
+    return Observable.of( this.data.filter(e => {if(e.id == id)return e})[0]);
```
Comment (chraker) [inline frontend/web/src/app/services/physical-print/physical-print-mock.service.ts]: MAJOR == should be === [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS1440)

File: frontend/web/src/app/services/physical-print/physical-print-mock.service.ts [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+  }
+  delete(id: number) {
+    this.data = this.data.filter(e => {if(e.id != id)return e});
```
Comment (chraker) [inline frontend/web/src/app/services/physical-print/physical-print-mock.service.ts]: MAJOR != should be !== [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS1440)

File: frontend/web/src/app/components/customer-detail/customer-detail.component.ts [hunk 10]
```diff
@@ -1,28 +1,29 @@
...
   selector: 'app-customer-detail',
   templateUrl: './customer-detail.component.html',
   styleUrls: ['./customer-detail.component.scss'],
   providers: [CustomerService, ErrorService]
 })
-export class CustomerDetailComponent implements OnInit {
+export class CustomerDetailComponent implements OnInit, OnChanges {
+  @Input('customer') customer: Customer = null;
+  @Input('nav') nav: boolean = true;
+  @Input('creating') creating: boolean = false;
```
Comment (chraker) [inline frontend/web/src/app/components/customer-detail/customer-detail.component.ts]: MINOR Type boolean trivially inferred from a boolean literal, remove type annotation [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS3257)

**Comment by chraker** [comment]
SonarQube analysis reported 10 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: frontend/web/src/app/components/customer-list/customer-list.component.ts (hunk 1 above)
Comment (chraker) [inline frontend/web/src/app/components/customer-list/customer-list.component.ts]: MINOR Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead. [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS4137)

File: frontend/web/src/app/components/customer-list/customer-list.component.ts (hunk 2 above)
Comment (chraker) [inline frontend/web/src/app/components/customer-list/customer-list.component.ts]: MAJOR == should be === [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS1440)

File: frontend/web/src/app/model/order.ts (hunk 3 above)
Comment (chraker) [inline frontend/web/src/app/model/order.ts]: CRITICAL if statements must be braced [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS121)

File: frontend/web/src/app/model/order.ts (hunk 4 above)
Comment (chraker) [inline frontend/web/src/app/model/order.ts]: CRITICAL if statements must be braced [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS121)

File: frontend/web/src/app/model/order.ts (hunk 5 above)
Comment (chraker) [inline frontend/web/src/app/model/order.ts]: CRITICAL else statements must be braced [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS121)

File: frontend/web/src/app/services/physical-part/physical-part-mock.service.ts (hunk 6 above)
Comment (chraker) [inline frontend/web/src/app/services/physical-part/physical-part-mock.service.ts]: MAJOR != should be !== [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS1440)

File: frontend/web/src/app/services/physical-part/physical-part-mock.service.ts (hunk 7 above)
Comment (chraker) [inline frontend/web/src/app/services/physical-part/physical-part-mock.service.ts]: MAJOR == should be === [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS1440)

File: frontend/web/src/app/services/physical-print/physical-print-mock.service.ts (hunk 8 above)
Comment (chraker) [inline frontend/web/src/app/services/physical-print/physical-print-mock.service.ts]: MAJOR == should be === [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS1440)

File: frontend/web/src/app/services/physical-print/physical-print-mock.service.ts (hunk 9 above)
Comment (chraker) [inline frontend/web/src/app/services/physical-print/physical-print-mock.service.ts]: MAJOR != should be !== [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS1440)

File: frontend/web/src/app/components/customer-detail/customer-detail.component.ts (hunk 10 above)
Comment (chraker) [inline frontend/web/src/app/components/customer-detail/customer-detail.component.ts]: MINOR Type boolean trivially inferred from a boolean literal, remove type annotation [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS3257)

**Comment by chraker** [comment]
SonarQube analysis reported 10 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: frontend/web/src/app/components/customer-list/customer-list.component.ts (hunk 1 above)
Comment (chraker) [inline frontend/web/src/app/components/customer-list/customer-list.component.ts]: MINOR Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead. [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS4137)

**Comment by chraker** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: frontend/web/src/app/components/customer-list/customer-list.component.ts [hunk 11]
```diff
@@ -88,27 +89,27 @@ export class CustomerListComponent implements OnInit, AfterViewInit {
...
+    (this.table as any).bootstrapTable('destroy');
+    (this.table as any).bootstrapTable(
       {
-        "data": data,
-        "columns": columns,
-        "sortStable": true,
-        "sortName": "name"
-      }
+        data: data,
+        columns: columns,
```
Comment (chraker) [inline frontend/web/src/app/components/customer-list/customer-list.component.ts]: MINOR Expected property shorthand in object literal ('{columns}'). [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS3498)

File: frontend/web/src/app/components/customer-list/customer-list.component.ts [hunk 12]
```diff
@@ -88,27 +89,27 @@ export class CustomerListComponent implements OnInit, AfterViewInit {
...
-    (<any>this.table).bootstrapTable(
+    (this.table as any).bootstrapTable('destroy');
+    (this.table as any).bootstrapTable(
       {
-        "data": data,
-        "columns": columns,
-        "sortStable": true,
-        "sortName": "name"
-      }
+        data: data,
```
Comment (chraker) [inline frontend/web/src/app/components/customer-list/customer-list.component.ts]: MINOR Expected property shorthand in object literal ('{data}'). [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS3498)

File: frontend/web/src/app/components/digital-part-list/digital-part-list.component.ts [hunk 13]
```diff
@@ -37,73 +37,73 @@ export class DigitalPartListComponent implements OnInit, AfterViewInit {
...
-    (<any>this.table).bootstrapTable(
+    (this.table as any).bootstrapTable('destroy');
+    (this.table as any).bootstrapTable(
       {
-        "data": data,
-        "columns": columns,
-        "sortStable": true,
-        "sortName": "name"
-      }
+        data: data,
```
Comment (chraker) [inline frontend/web/src/app/components/digital-part-list/digital-part-list.component.ts]: MINOR Expected property shorthand in object literal ('{data}'). [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS3498)

File: frontend/web/src/app/components/digital-part-list/digital-part-list.component.ts [hunk 14]
```diff
@@ -37,73 +37,73 @@ export class DigitalPartListComponent implements OnInit, AfterViewInit {
...
+        columns: columns,
```
Comment (chraker) [inline frontend/web/src/app/components/digital-part-list/digital-part-list.component.ts]: MINOR Expected property shorthand in object literal ('{columns}'). [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS3498)

**Comment by chraker** [comment]
SonarQube analysis reported 7 issues
* ![MINOR][MINOR] 7 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [app.component.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/dd1a54192fa6b45b34edd64cf1171c37df169248/frontend/web/src/app/app.component.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)
1. ![MINOR][MINOR] [app.module.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/dd1a54192fa6b45b34edd64cf1171c37df169248/frontend/web/src/app/app.module.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)
1. ![MINOR][MINOR] [http-client.service.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/dd1a54192fa6b45b34edd64cf1171c37df169248/frontend/web/src/app/services/http/http-client.service.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: frontend/web/src/app/components/customer-list/customer-list.component.ts (hunk 11 above)
Comment (chraker) [inline frontend/web/src/app/components/customer-list/customer-list.component.ts]: MINOR Expected property shorthand in object literal ('{columns}'). [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS3498)

File: frontend/web/src/app/components/customer-list/customer-list.component.ts (hunk 12 above)
Comment (chraker) [inline frontend/web/src/app/components/customer-list/customer-list.component.ts]: MINOR Expected property shorthand in object literal ('{data}'). [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS3498)

File: frontend/web/src/app/components/digital-part-list/digital-part-list.component.ts (hunk 13 above)
Comment (chraker) [inline frontend/web/src/app/components/digital-part-list/digital-part-list.component.ts]: MINOR Expected property shorthand in object literal ('{data}'). [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS3498)

File: frontend/web/src/app/components/digital-part-list/digital-part-list.component.ts (hunk 14 above)
Comment (chraker) [inline frontend/web/src/app/components/digital-part-list/digital-part-list.component.ts]: MINOR Expected property shorthand in object literal ('{columns}'). [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS3498)

**Comment by chraker** [comment]
SonarQube analysis reported 7 issues
* ![MINOR][MINOR] 7 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [app.component.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/df7ffb936abff0bbded1712c07babca6dbdce6cc/frontend/web/src/app/app.component.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)
1. ![MINOR][MINOR] [app.module.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/df7ffb936abff0bbded1712c07babca6dbdce6cc/frontend/web/src/app/app.module.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)
1. ![MINOR][MINOR] [http-client.service.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/df7ffb936abff0bbded1712c07babca6dbdce6cc/frontend/web/src/app/services/http/http-client.service.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by chraker** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [app.component.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/f796e763ea65459306b785ae4b1a0927676673cd/frontend/web/src/app/app.component.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)
1. ![MINOR][MINOR] [app.module.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/f796e763ea65459306b785ae4b1a0927676673cd/frontend/web/src/app/app.module.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)
1. ![MINOR][MINOR] [http-client.service.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/f796e763ea65459306b785ae4b1a0927676673cd/frontend/web/src/app/services/http/http-client.service.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by chraker** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [app.component.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/e40964230a1662a7c775b06e13a53c6be694a6d9/frontend/web/src/app/app.component.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)
1. ![MINOR][MINOR] [app.module.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/e40964230a1662a7c775b06e13a53c6be694a6d9/frontend/web/src/app/app.module.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)
1. ![MINOR][MINOR] [http-client.service.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/e40964230a1662a7c775b06e13a53c6be694a6d9/frontend/web/src/app/services/http/http-client.service.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by chraker** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [app.component.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/733ade6eedc846c1dbb7675c23d07e95b03e761c/frontend/web/src/app/app.component.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)
1. ![MINOR][MINOR] [app.module.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/733ade6eedc846c1dbb7675c23d07e95b03e761c/frontend/web/src/app/app.module.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)
1. ![MINOR][MINOR] [http-client.service.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/733ade6eedc846c1dbb7675c23d07e95b03e761c/frontend/web/src/app/services/http/http-client.service.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by chraker** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [app.component.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/733ade6eedc846c1dbb7675c23d07e95b03e761c/frontend/web/src/app/app.component.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)
1. ![MINOR][MINOR] [app.module.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/733ade6eedc846c1dbb7675c23d07e95b03e761c/frontend/web/src/app/app.module.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)
1. ![MINOR][MINOR] [http-client.service.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/733ade6eedc846c1dbb7675c23d07e95b03e761c/frontend/web/src/app/services/http/http-client.service.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by chraker** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [app.component.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/733ade6eedc846c1dbb7675c23d07e95b03e761c/frontend/web/src/app/app.component.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)
1. ![MINOR][MINOR] [app.module.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/733ade6eedc846c1dbb7675c23d07e95b03e761c/frontend/web/src/app/app.module.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)
1. ![MINOR][MINOR] [http-client.service.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/733ade6eedc846c1dbb7675c23d07e95b03e761c/frontend/web/src/app/services/http/http-client.service.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by eineving — COMMENTED** [review]
The none frontend parts look good, someone with mopre frontend knowledge should take a look

**Comment by chraker** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [app.component.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/acadd3b332554d7ae77cd678d8b96c3803522473/frontend/web/src/app/app.component.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)
1. ![MINOR][MINOR] [app.module.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/acadd3b332554d7ae77cd678d8b96c3803522473/frontend/web/src/app/app.module.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)
1. ![MINOR][MINOR] [http-client.service.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/acadd3b332554d7ae77cd678d8b96c3803522473/frontend/web/src/app/services/http/http-client.service.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by chraker** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [app.component.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/f9e6785d52b150efa0b2b82089bde11256b9392b/frontend/web/src/app/app.component.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)
1. ![MINOR][MINOR] [app.module.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/f9e6785d52b150efa0b2b82089bde11256b9392b/frontend/web/src/app/app.module.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)
1. ![MINOR][MINOR] [http-client.service.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/f9e6785d52b150efa0b2b82089bde11256b9392b/frontend/web/src/app/services/http/http-client.service.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by chraker** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [app.component.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/ff3ffe001b88b7ed1f8ff5f766d9f87e569167c2/frontend/web/src/app/app.component.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)
1. ![MINOR][MINOR] [app.module.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/ff3ffe001b88b7ed1f8ff5f766d9f87e569167c2/frontend/web/src/app/app.module.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)
1. ![MINOR][MINOR] [http-client.service.ts](https://github.com/SoftwareEvolutionProject/2017Group1/blob/ff3ffe001b88b7ed1f8ff5f766d9f87e569167c2/frontend/web/src/app/services/http/http-client.service.ts): file should end with a newline [rule](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS113)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by mlonn — APPROVED** [review]
No it is okay!

**Pull request merged**